server/src/db/lifecycle/compact.rs

```diff
@@ -58,6 +62,8 @@ pub(crate) fn compact_chunks(
                 .map(|prev_last| prev_last.max(candidate_last))
                 .or(Some(candidate_last));
 
+            delete_predicates = chunk.delete_predicates();
```

Comment:
As written this code appears to use the delete predicates for the last chunk (rather than all the delete predicates for all the chunks) when compacting the list. 

Perhaps the code should be appending the delete predicates of the chunk (rather than overwriting them)? Something like

```suggestion
            delete_predicates.append(chunk.delete_predicates().clone());
```

Reply:
This is a great catch